src/main/java/bsh/Interpreter.java

```diff
@@ -1192,6 +1192,18 @@ private String getBshPrompt()
         }
     }
 
+    /**
+     * Interpreter calls <code>prompt()</code> when the prompt shall be displayed
+     * on console. The default implementation is that prompt is just printed to
+     * the Console/Interpreter's output stream, but subclasses with a more
+     * sophisticated prompt handling (like terminals) can hook here their logic.
+     *
+     * @param prompt the prompt to me displayed
+     */
+    protected void prompt(String prompt) {
+        print(prompt);
+    }
```

Comment:
Can we make prompt a ConsoleInteface method?

And implement this method similar to error and print:

```
if ( null != console ) {
   console.prompt(prompt);
else
   print(prompt);
}
```

This way the console implementation, ie for jline, can implement its prompt method as it sees fit.

Reply:
What I was thinking is an inner class `DefaultConsole` or something, which implements `ConsoleInteface` and holds the references to `in`, `out` and `err` with the interpreter getIn, getOut, getErr, print, println, error delegating to the innerclass instance referenced at console.

If you want to start on that this would be awesome!!! Really needed!

But if you are pressed for time just the `prompt` method as a default method on interface for now would be great and gets this PR committed.